pgmpy/prediction/BaseCausalPrediction.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator
+
+
+class BaseCausalPrediction(BaseEstimator):
+    """
+    Base class for causal prediction models in pgmpy. Provides common
+    functionality for preparing and validating feature dataframes.
+    """
+
+    def _prepare_feature_df(self, X, required_features=None) -> pd.DataFrame:
+        """
+        Convert input (either numpy array or dataframe) to a DataFrame and
+        validate that column names exactly match DAG variables.
+
+        If a numpy array is provided, it is converted to a DataFrame with
+        range index column names (0, 1, ..., n_features-1).
+
+        Parameters
+        ----------
+        X : array-like or DataFrame
+            Input features.
+
+        required_features : list[int] or None
+            Column indices expected from the DAG. If None, uses
+            `self.feature_columns_fit_`.
```

Comment:
Let's make it required, as that would be much more explicit. In the current form, it would be very difficult to guess what the default is without looking at this docstring.